metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -138,6 +139,15 @@ public List<AuthorizationResult> authorize(
         return results;
     }
```

Comment:
The purpose of doing this is to avoid loading the volatile multiple times. Each time we load a volatile, it is expensive because it requires an interlocked instruction.

Reply:
It also ensures a consistent snapshot is used for the duration of the method, right? Otherwise, one could get an updated version of the field for some of the authorize calls.